### PR TITLE
fixes #4239, pretty printing for single arg infix op terms no longer displays "error"

### LINKED ISSUE #4239: Rendering an infix function created using `cases` produces invalid code
Version: release/M5b

A somewhat weirdly defined (but presumably valid) infix function:
```
(>>>>) : Nat -> Nat -> ()
(>>>>) n = cases
  _ -> bug ""
 ```
 
 After adding it and viewing it:
 ```
 @unison/httpserver/main> view >>>>

  (>>>>) : Nat -> Nat -> ()
  error = cases _ -> bug ""
 ```
 
 (Note the `error` name after the type)
 
 Example on share:
 https://share.unison-lang.org/@unison/httpserver/code/releases/0.3.0/latest/terms/Routes/%3E%3E%3E

### PR DESCRIPTION
**Pull request by kylegoetz**
## Overview

If one were to add this code:

```
(>>>>) : Nat -> Nat -> ()
(>>>>) n = cases
  _ -> bug ""
```

and then view it, one would see this output:

```
(>>>>) : Nat -> Nat -> ()
 error = cases _ -> bug ""
```

Observe that the function name+params is just "error" now.

The view should instead output

```
(>>>>) : Nat -> Nat -> ()
(>>>>) n = cases _ -> bug ""
```

This closes issue #4239.

## Implementation notes

In `TermPrinter.hs`, in some codepaths, the parameters of an infix function are parsed such that if there are two parameters provided, the term definition's LHS is "(symop) p1 p2" and otherwise rendered as the string "error".

I added code to handle the case where only one parameter is provided in the LHS of the term definition/binding.

## Interesting/controversial decisions

There were two ways I could've gone about this. One, informed by the way the pretty printer renders the definition of an infix function:

```
a >>>> b = ...
```

As such, I tried to pretty print this first:

```
n >>> _ = ...
```

However, this is a syntax error because `_` is not a valid param it seems.

So instead I chose to render as a prefix symop with one param, `(>>>>) n = ...` instead.

## Test coverage

I added a transcript test that will ensure that the formerly-buggy code produces now

```
(>>>>) n = cases _ -> bug ""
```

instead of the former, where "error" was in place of "(>>>>) n".

### CODE REVIEW

**Comment by kylegoetz** [comment]
Curiously, this bug does not occur for all symop term definitions with just one provided param. I tried with something like

```
(>>>>) n = let
  a = 5
  b = 2
  ()
```

and all that happened was "let" was stripped out. The function name + params remained correct.

**Comment by kylegoetz** [comment]
It's unclear to me why the ormolu container in the CI stage failed. Or, rather, whether something about my PR caused the failure, which is some kind of permissions thing?

```
remote: Permission to unisonweb/unison.git denied to github-actions[bot].
[423](https://github.com/unisonweb/unison/actions/runs/5827085920/job/15802381729?pr=4256#step:5:424)
fatal: unable to access 'https://github.com/unisonweb/unison/': The requested URL returned error: 403
[424](https://github.com/unisonweb/unison/actions/runs/5827085920/job/15802381729?pr=4256#step:5:425)
Error: Invalid status code: 128
[425](https://github.com/unisonweb/unison/actions/runs/5827085920/job/15802381729?pr=4256#step:5:426)
    at ChildProcess.<anonymous> (/home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4/index.js:17:19)
[426](https://github.com/unisonweb/unison/actions/runs/5827085920/job/15802381729?pr=4256#step:5:427)
    at ChildProcess.emit (node:events:527:28)
[427](https://github.com/unisonweb/unison/actions/runs/5827085920/job/15802381729?pr=4256#step:5:428)
    at maybeClose (node:internal/child_process:1092:16)
[428](https://github.com/unisonweb/unison/actions/runs/5827085920/job/15802381729?pr=4256#step:5:429)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5) {
[429](https://github.com/unisonweb/unison/actions/runs/5827085920/job/15802381729?pr=4256#step:5:430)
  code: 128
[430](https://github.com/unisonweb/unison/actions/runs/5827085920/job/15802381729?pr=4256#step:5:431)
}
[431](https://github.com/unisonweb/unison/actions/runs/5827085920/job/15802381729?pr=4256#step:5:432)
Error: Invalid status code: 128
[432](https://github.com/unisonweb/unison/actions/runs/5827085920/job/15802381729?pr=4256#step:5:433)
    at ChildProcess.<anonymous> (/home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4/index.js:17:19)
[433](https://github.com/unisonweb/unison/actions/runs/5827085920/job/15802381729?pr=4256#step:5:434)
    at ChildProcess.emit (node:events:527:28)
[434](https://github.com/unisonweb/unison/actions/runs/5827085920/job/15802381729?pr=4256#step:5:435)
    at maybeClose (node:internal/child_process:1092:16)
[435](https://github.com/unisonweb/unison/actions/runs/5827085920/job/15802381729?pr=4256#step:5:436)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
```

File: parser-typechecker/src/Unison/Syntax/TermPrinter.hs
Comment (pchiusano) [inline parser-typechecker/src/Unison/Syntax/TermPrinter.hs]:
```suggestion
              [x] ->
```

**Review by pchiusano — APPROVED** [review]
Looks great! I made one very minor style change. I'll merge once CI passes.

I'm not sure what's going on with the Ormolu action, maybe @aryairani knows?

**Comment by ChrisPenner** [comment]
We're having some trouble with our nix CI job right now, but it's not anything that needs to block this from being merged 👍🏼

**Comment by kylegoetz** [comment]
Thanks for figuring this out. I hadn't gotten around to the alternative solutions for merging that had been proposed!

**Pull request merged**
